backends/dpdk/dpdkVarCollector.cpp

```diff
@@ -0,0 +1,31 @@
+/*
+Copyright 2020 Intel Corp.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#include "dpdkVarCollector.h"
+
+namespace DPDK {
+cstring DpdkVariableCollector::get_next_tmp() {
+  std::ostringstream out;
+  out << "backend_tmp" << next_tmp_id;
```

Comment:
This does not look safe; such a variable could appear in the program.
The reference map has a method to generate a fresh identifier that is guaranteed not to occur in the program.
Or you should use a name that is syntactically legal in dpdk but not in P4.

Reply:
@gordonwucn please take a look.

Reply:
@hanw this naming method is not safe. Can you use the method that Mihai mentioned to generate the names that will not cause conflicts?

Reply:
fixed.